integrationtests/volume_test.go

```diff
@@ -0,0 +1,305 @@
+	runNegativeFormatVolumeRequest(t, client, "-1")
+
+	// Resize volume negative tests
+	runNegativeResizeVolumeRequest(t, client, "", 2*1024*1024)
+	runNegativeResizeVolumeRequest(t, client, "-1", 2*1024*1024)
+
+	// Mount volume negative tests
+	runNegativeMountVolumeRequest(t, client, "", "")
+	runNegativeMountVolumeRequest(t, client, "-1", "")
+
+	// Dismount volume negative tests
+	runNegativeDismountVolumeRequest(t, client, "", "")
+	runNegativeDismountVolumeRequest(t, client, "-1", "")
+}
+
+func negativeDiskTests(t *testing.T) {
+	var client *v1alpha1client.Client
+	var err error
+
+	if client, err = v1alpha1client.NewClient(); err != nil {
+		t.Fatalf("Client new error: %v", err)
+	}
+	defer client.Close()
+
+	// Empty disk Id test
+	runNegativeListVolumeRequest(t, client, "")
+	// Negative disk id test
+	runNegativeListVolumeRequest(t, client, "-1")
+}
+
+func simpleE2e(t *testing.T) {
+	var client *v1alpha1client.Client
+	var err error
+
+	if client, err = v1alpha1client.NewClient(); err != nil {
+		t.Fatalf("Client new error: %v", err)
+	}
+	defer client.Close()
+
+	s1 := rand.NewSource(time.Now().UTC().UnixNano())
+	r1 := rand.New(s1)
+
+	testPluginPath := fmt.Sprintf("C:\\var\\lib\\kubelet\\plugins\\testplugin-%d.csi.io\\", r1.Intn(100))
+	mountPath := fmt.Sprintf("%smount-%d", testPluginPath, r1.Intn(100))
+	vhdxPath := fmt.Sprintf("%sdisk-%d.vhdx", testPluginPath, r1.Intn(100))
+
+	defer diskCleanup(t, vhdxPath, mountPath, testPluginPath)
+	diskNum := diskInit(t, vhdxPath, mountPath, testPluginPath)
+
+	listRequest := &v1alpha1.ListVolumesOnDiskRequest{
+		DiskId: diskNum,
+	}
+	listResponse, err := client.ListVolumesOnDisk(context.TODO(), listRequest)
+	if err != nil {
+		t.Fatalf("List response: %v", err)
+	}
+
+	volumeIDsLen := len(listResponse.VolumeIds)
+	if volumeIDsLen != 1 {
+		t.Fatalf("Number of volumes not equal to 1: %d", volumeIDsLen)
+	}
+	volumeID := listResponse.VolumeIds[0]
+
+	isVolumeFormattedRequest := &v1alpha1.IsVolumeFormattedRequest{
+		VolumeId: volumeID,
+	}
+	isVolumeFormattedResponse, err := client.IsVolumeFormatted(context.TODO(), isVolumeFormattedRequest)
+	if err != nil {
+		t.Fatalf("Is volume formatted request error: %v", err)
+	}
+	if isVolumeFormattedResponse.Formatted {
+		t.Fatal("Volume formatted. Unexpected !!")
+	}
+
+	formatVolumeRequest := &v1alpha1.FormatVolumeRequest{
+		VolumeId: volumeID,
+	}
+	_, err = client.FormatVolume(context.TODO(), formatVolumeRequest)
+	if err != nil {
+		t.Fatalf("Volume format failed. Error: %v", err)
+	}
+
+	isVolumeFormattedResponse, err = client.IsVolumeFormatted(context.TODO(), isVolumeFormattedRequest)
+	if err != nil {
+		t.Fatalf("Is volume formatted request error: %v", err)
+	}
+	if !isVolumeFormattedResponse.Formatted {
+		t.Fatal("Volume should be formatted. Unexpected !!")
+	}
+
+	resizeVolumeRequest := &v1alpha1.ResizeVolumeRequest{
+		VolumeId: volumeID,
+		// Resize from 5G to 2G
+		Size: 2 * 1024 * 1024 * 1024,
+	}
+
```

Comment:
Will it be possible to validate the resized volume's size to appropriate size did happen through powershell here?

Reply:
Will add.